### PR TITLE
feat(web): remove the dependency on `package:js` in favor of `dart:js_interop`

### PR DESCRIPTION
**Pull request by Lyokone**
## Description

Also removing the `web` package when it wasn't used
Removed tests that were generating mocks since those are not compatible with the new interop and were passthrough tests.

## Related Issues

#12466 

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/firebase/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Review by TarekkMA — CHANGES_REQUESTED** [review]
Missing `js_interop` import on some files, and I guess firebase_core_web isn't migrated to js_interop yet

File: packages/firebase_database/firebase_database/example/lib/main.dart [hunk 1]
```diff
@@ -34,6 +34,16 @@ Future<void> main() async {
...
 
+  final database = FirebaseDatabase.instance;
+
+  await database.ref('tests/flutterfire').set(0);
+
+  final ref = database.ref('tests/flutterfire');
+
+  final snapshot = await ref.get();
+  print(snapshot.key);
+  print(snapshot.value);
```
Comment (russellwheatley) [inline packages/firebase_database/firebase_database/example/lib/main.dart:45]: presumably this can go now

**Review by russellwheatley — CHANGES_REQUESTED** [review]

**Review by exaby73 — APPROVED** [review]

File: tests/integration_test/e2e_test.dart [hunk 2]
```diff
@@ -38,9 +38,9 @@ void main() {
   group('FlutterFire', () {
     if (kIsWeb || !Platform.isWindows) {
       firebase_core.main();
+      firebase_auth.main();
       firebase_database.main();
       firebase_crashlytics.main();
-      firebase_auth.main();
```
Comment (TarekkMA) [inline tests/integration_test/e2e_test.dart:43]: any reason for the change?

**Review by TarekkMA — APPROVED** [review]

File: tests/integration_test/e2e_test.dart (hunk 2 above)
Comment (Lyokone) [reply tests/integration_test/e2e_test.dart:43]: We have add a race condition with auth for a bit of time now, and I tried this out of curiosity and seems to work better ... Might be just placebo tho 😅

**Comment by kevmoo** [comment]
Can we get this merged?

**Review by russellwheatley — APPROVED** [review]

**Comment by Lyokone** [comment]
@kevmoo Merged, will be part of next release that should be out today :)

**Pull request merged**